src/Lazy/takeRight.ts

```diff
@@ -0,0 +1,149 @@
+}
+
+function* arrayTakeRight<A>(length: number, arr: Array<A>) {
+  const index = arr.length - length;
+  for (let i = index; i < arr.length; i++) {
+    if (arr[i]) yield arr[i];
+  }
+}
+
+function* sync<A>(length: number, iterable: Iterable<A>): IterableIterator<A> {
+  const arr = toArray(iterable);
+  const index = arr.length - length;
+  for (let i = index; i < arr.length; i++) {
+    if (arr[i]) yield arr[i];
+  }
+}
+
+async function* asyncSequential<T>(length: number, iterable: AsyncIterable<T>) {
+  const arr = await toArray(iterable);
+  const index = arr.length - length;
+  for (let i = index; i < arr.length; i++) {
+    if (arr[i]) {
+      yield arr[i];
+    }
+  }
+}
+
+function async<A>(
+  length: number,
+  iterable: AsyncIterable<A>,
+): AsyncIterableIterator<A> {
+  let iterator: AsyncIterator<A>;
+  return {
+    [Symbol.asyncIterator]() {
+      return this;
+    },
+    next(_concurrent: any) {
+      if (iterator === undefined) {
+        iterator = isConcurrent(_concurrent)
+          ? asyncSequential(length, concurrent(_concurrent.length, iterable))
+          : asyncSequential(length, iterable);
+      }
+
+      return iterator.next(_concurrent);
+    },
+  };
+}
+
+/**
+ * Returns Iterable/AsyncIterable that taken the last argument `l` values from iterable
+ *
+ * @example
+ * ```ts
+ * const iter = takeRight(2, [0, 1, 2, 3, 4, 5, 6]);
+ * iter.next() // {done:false, value: 5}
+ * iter.next() // {done:false, value: 6}
+ * iter.next() // {done:true, value: undefined}
+ *
+ * // with pipe
+ * pipe(
+ *  [0, 1, 2, 3, 4, 5, 6],
+ *  takeRight(2),
+ *  toArray,
+ * ); // [5, 6]
+ *
+ * await pipe(
+ *  Promise.resolve([0, 1, 2, 3, 4, 5, 6]),
+ *  takeRight(2),
+ *  toArray,
+ * ); // [5, 6]
+ *
+ * // with toAsync
+ * await pipe(
+ *  [Promise.resolve(0), Promise.resolve(1), Promise.resolve(2),
+ *   Promise.resolve(3), Promise.resolve(4), Promise.resolve(5), Promise.resolve(6)],
+ *  toAsync,
+ *  takeRight(2),
+ *  toArray,
+ * ); // [5, 6]
+ * ```
+ *
+ * {@link https://codesandbox.io/s/fxts-take-372bs | Try It}
+ *
+ * see {@link https://fxts.dev/docs/pipe | pipe}, {@link https://fxts.dev/docs/toAsync | toAsync},
+ * {@link https://fxts.dev/docs/toArray | toArray}
+ */
+// prettier-ignore
+function takeRight<A>(
+  l: number,
+  iterable: Iterable<A>
+): IterableIterator<A>;
+
+function takeRight<A>(
+  l: number,
+  iterable: AsyncIterable<A>,
+): AsyncIterableIterator<A>;
+
+function takeRight<A extends Iterable<unknown> | AsyncIterable<unknown>>(
+  l: number,
+): (iterable: A) => ReturnIterableIteratorType<A>;
+
+function takeRight<A extends Iterable<unknown> | AsyncIterable<unknown>>(
+  l: number,
+  iterable?: A,
+):
+  | IterableIterator<IterableInfer<A>>
+  | AsyncIterableIterator<IterableInfer<A>>
+  | ((iterable: A) => ReturnIterableIteratorType<A>) {
+  if (iterable === undefined) {
+    return (iterable: A) => {
+      return takeRight(l, iterable as any) as ReturnIterableIteratorType<A>;
+    };
+  }
+
```

Comment:
Since there is a logic that evaluates everything with toArray, it seems to be necessary to check the correct argument.

```ts
  if (length < 1) {
    throw new TypeError("'length' must be greater than 1");
  }
```

Reply:
Thanks for your kind comment!
I added this logic and please check it if I'm correct!